sklearn/cluster/k_means_.py

```diff
@@ -845,7 +845,7 @@ def _check_fit_data(self, X):
 
     def _check_test_data(self, X):
         X = check_array(X, accept_sparse='csr', dtype=FLOAT_DTYPES,
-                        warn_on_dtype=True)
+                        warn_on_dtype=False)
```

Comment:
`warn_on_dtype=False` is already the default. To make the code less verbose and more consistent with the rest of the code base, please omit it.